File: Data Structures & Algorithms/house-robber/submission-11.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size() == 1) return nums[0];

        int a = 0;
        int b = 0;

        for(int n : nums){
            int tmp = max(a + n, b);
            a = b;
            b = tmp;
        }

        return b;
        
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-11.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        s = set(nums)
        res = 0

        for val in s:
            if val - 1 not in s:
                tmp = val
                x = 0
                while tmp in s:
                    x += 1
                    tmp += 1
                    res = max(res, x)
        return res
        